coral-spark/src/main/java/com/linkedin/coral/spark/utils/RelDataTypeToHiveTypeStringConverter.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rel.type.RelRecordType;
+import org.apache.calcite.sql.type.ArraySqlType;
+import org.apache.calcite.sql.type.MapSqlType;
+
+
+/**
+ * Transforms a RelDataType to a Hive type string such that it is parseable and semantically correct.
+ */
+public class RelDataTypeToHiveTypeStringConverter {
+  private RelDataTypeToHiveTypeStringConverter() {
+  }
+
+  /**
+   * @param relDataType a given RelDataType
+   * @return a syntactically and semantically correct Hive type string for relDataType
+   */
+  public static String buildHiveTypeString(RelDataType relDataType) {
+    switch (relDataType.getSqlTypeName()) {
+      case ROW:
+        return buildStructDataTypeString((RelRecordType) relDataType);
+      case ARRAY:
+        return buildArrayDataTypeString((ArraySqlType) relDataType);
+      case MAP:
+        return buildMapDataTypeString((MapSqlType) relDataType);
+      case INTEGER:
+        return "int";
+      case SMALLINT:
+        return "smallint";
+      case TINYINT:
+        return "tinyint";
+      case BIGINT:
+        return "bigint";
+      case DOUBLE:
+        return "double";
+      case REAL:
+      case FLOAT:
+        return "float";
+      case BOOLEAN:
+        return "boolean";
+      case CHAR:
```

Comment:
I remember Hive has `char` type, can you check it?

Reply:
Thanks, yes I missed it. Just added it back